### ISSUE TITLE
Add LzwReader Support for .Z Archives Using LzwStream

### ISSUE THREAD

**Issue by adanddev** [adamhathcock/sharpcompress#880]
Please consider to add support for reading .Z compressed archives by implementing an LzwReader (name can be different) that uses `LzwStream` for decompression. This new reader should follow the design patterns of existing readers, such as GZipReader, so that it can be created through the standard factory methods in SharpCompress.

**Comment by Erior** [comment]
Don't we already support decompress .Z, as in test file Tar.tar.Z ?
or are you looking for creating .Z files?

**Comment by adanddev** [comment]
Yes it there is support for extracting .Z files when they are part of a `.tar.Z` archive (e.g., Tar.tar.Z), but what I'm referring to is the ability to extract standalone .Z files, such as file.Z, that do not contain a tarball inside. These files are simply compressed using LZW without any additional archive format.

Currently, it seems we lack support for directly decompressing such standalone .Z files. Adding this capability would make SharpCompress more versatile for handling legacy compression formats.

**Comment by neumatho** [comment]
You can use e.g. XZStream() or some of the others. I do that to get this functionality in my project. To be sure the file is in the right format, you can use XZStream.IsXZStream().
